storage/rocksdb/CMakeLists.txt

```diff
@@ -123,6 +123,12 @@ INCLUDE_DIRECTORIES(
 ADD_DEFINITIONS(-DROCKSDB_PLATFORM_POSIX -DROCKSDB_LIB_IO_POSIX -DZLIB -DLZ4
   -DZSTD -DROCKSDB_SUPPORT_THREAD_LOCAL)
 
+IF (NOT "$ENV{WITH_TBB}" STREQUAL "")
+  SET(rocksdb_static_libs ${rocksdb_static_libs}
+  $ENV{WITH_TBB}/libtbb${PIC_EXT}.a)
+  ADD_DEFINITIONS(-DTBB)
```

Comment:
rocksdb_use_clock_cache = TRUE requires this, so now we must update build deps, Jenkins, everything, to get TBB in the picture, at some correct version.

rocksdb_use_clock_cache is also missing any tests except for the sysvar test that does not exercise its "TRUE" value.

Reply:
Yeah, I was actually suspicious of taking in that(those) patch(s) at all. TBB though seems is carried with glibc as evidenced that this actually did build on all of our builders. I don't think there is anything really to do there other than maybe also add conditional compilation of the option and its use in MyRocks and maybe fail to cmake MyRocks if it is missing.

Agreed that there is missing test but I'm really unsure of what can be validated other than the server can start with rocksdb_use_clock_cache=on and select @@global.rocksdb_use_clock_cache.

I am inclined to remove this entirely from PS but I suspect that this will be another of those edge case perf improvements that will be complained about if it is missing.

Reply:
This is more like a "SSE4.2" situation in that a build w/o dep simply results in missing functionality, not a build failure, and a runtime error only. Thus builders succeeding does not tell anything about TBB. That missing test would, which can be as simple as possible (start an instance, run some RW workload) to run the code paths.

Fine if we can remove it but for all I know clock cache could be the future and not an edge case. Thus I don't think we can ignore it that easily, unfortunately.

Reply:
Uhm CMake diff looks the same as in the previous iteration, am I missing something? I wrote the above without seeing the new test yet, but based on the code review. The test passing looks wrong to me, please see below a comment re. test.

Reply:
Crap, somehow, my revisions to CMakeLists.txt did not get pushed. I need to do this one over. The basic idea is to demand that TBB is available and warn/error on cmake based on WITH_ROCKSDB=0|1|FORCE

Reply:
I'm just going to pull this entirely